File: 2022/day06.py
with open("input_files/day06", "r") as f:
    data = f.read()

fours = [set(data[i:i+4]) for i in range(len(data))]
fourteens = [set(data[i:i + 14]) for i in range(len(data))]

idx4 = next(i + 4 for i, window in enumerate(fours) if len(window) == 4)
idx14 = next(i + 14 for i, window in enumerate(fourteens) if len(window) == 14)

print(idx4)
print(idx14)

File: kotlin/src/main/kotlin/y2022/day06.kt
package y2022

import read

fun main() {
    val data = read("y2022/day06")
    val fours = data.windowed(4).map { it.toSet() }
    val fourteens = data.windowed(14).map { it.toSet() }

    println(fours.indexOfFirst { it.size == 4 } + 4)
    println(fourteens.indexOfFirst { it.size == 14 } + 14)
}
